api/filter_dimensions.go

```diff
@@ -221,3 +223,17 @@ func (api *FilterAPI) checkNewFilterDimension(ctx context.Context, name string,
 
 	return nil
 }
+
+func removeDuplicates(elements []string) []string {
```

Comment:
Maybe name could be more specific? removeDuplicateOptions

Worth adding a couple of unit tests for this function

Reply:
This is deliberately generic so that it can be reused for other things like dimensions.